src/test/java/pl/task/parkingmeter/controller/VehicleControllerTest.java

```diff
@@ -0,0 +1,384 @@
+
+        mockMvc.perform(delete("/{regNumber}", REG_NUMBER))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.regNumber").value(REG_NUMBER))
+                .andExpect(jsonPath("$.bill").value(valueRate.getValue())).andExpect(jsonPath("$.regNumber").value(REG_NUMBER))
+                .andExpect(jsonPath("$.paid").value(true))
+                .andExpect(jsonPath("$.payDate").isNotEmpty())
+                .andDo(print());
+    }
+
+    @Test
+    public void payForParkingWhenOkAndNotDefaultCurrency() throws Exception {
+
+        Vehicle vehicle = new Vehicle();
+        vehicle.setRegNumber(REG_NUMBER);
+        vehicle.setRunDate(LocalDateTime.now().minusHours(4).minusMinutes(30));
+        vehicle.setOwnerDisabled(false);
+        vehicle.setPaid(false);
+
+        Rates regularPLN = new Rates();
+        regularPLN.setCurrency("EUR");
+        regularPLN.setType("regular");
+
+        ValueRate valueRate = new ValueRate();
+        valueRate.setRates(regularPLN);
+        valueRate.setHours(7);
+        valueRate.setValue(BigDecimal.valueOf(4));
+        valueRate.setRates(regularPLN);
+
+        when(this.vehicleService.findVehicleByRegNumberAndPaidFalse(REG_NUMBER)).thenReturn(java.util.Optional.ofNullable(vehicle));
+
+        when(this.ratesService.findRatesByTypeAndCurrency("regular", "EUR")).thenReturn(regularPLN);
+
+        when(this.valueRateService.findValueRateByHoursAndRatesId(5, regularPLN)).thenReturn(valueRate);
+
+        mockMvc.perform(delete("/{regNumber}?currency=EUR", REG_NUMBER))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.regNumber").value(REG_NUMBER))
+                .andExpect(jsonPath("$.bill").value(valueRate.getValue())).andExpect(jsonPath("$.regNumber").value(REG_NUMBER))
+                .andExpect(jsonPath("$.paid").value(true))
+                .andExpect(jsonPath("$.payDate").isNotEmpty())
+                .andDo(print());
+    }
+
+    @Test
+    public void payForParkingButVehicleNotFound() throws Exception {
+
+        when(this.vehicleService.findVehicleByRegNumberAndPaidFalse(REG_NUMBER)).thenReturn(java.util.Optional.ofNullable(null));
+
+        mockMvc.perform(delete("/{regNumber}", REG_NUMBER))
+                .andExpect(status().is4xxClientError())
+                .andExpect(status().reason("Vehicle not found!"))
+                .andDo(print());
+    }
+
+    @Test
+    public void checkProfit() throws Exception {
+
+        String date = "2018-08-28";
+        LocalDateTime startDate = LocalDateTime.of(2018, 8, 28, 0, 0, 0);
+        LocalDateTime endDate = LocalDateTime.of(2018, 8, 28, 23, 59, 59);
+
+        Vehicle firstVehicle = new Vehicle();
+        firstVehicle.setRegNumber(REG_NUMBER);
+        firstVehicle.setPayDate(LocalDateTime.of(2018, 8, 28, 11, 11, 11));
+        firstVehicle.setCurrency("PLN");
+        firstVehicle.setBill(BigDecimal.valueOf(25));
+
+        Vehicle secondVehicle = new Vehicle();
```

Comment:
czy nie można by było setupu tych wszystkich obiektów umieścić w innym miejscy? tutaj strasznie zaciemniają test

Reply:
można je wrzucić do metod oznaczonych @Before lub @BeforeClass, wtedy dane będą się inicjowały przed każdym testem lub raz przed uruchomieniem testów

Reply:
ale to właśnie jest w metodzie `@BeforeClass` ;) czy jeszcze indziej można by je wynieść? może do innej klasy?